improver/utilities/cube_manipulation.py

```diff
@@ -81,6 +81,84 @@ def collapsed(cube: Cube, *args: Any, **kwargs: Any) -> Cube:
     return new_cube
 
 
+def aggregate(
+    cube: Cube,
+    dimensions: Union[str, List[str]] = "realization",
+    aggregation: str = "mean",
+    broadcast: bool = False,
+    new_name=None,
+):
+    """Aggregate a cube over the given dimensions.
+
+    Args:
+        cube:
+            Cube to aggregate
+        dimensions:
+            List of dimensions to aggregate; default is "realization"
+        aggregation:
+            One of "sum", "mean", "median", "std_dev", "min", "max";
+            default is "mean".
+        broadcast:
+            If True, broadcast result back to original dimensions.
+            Otherwise, return collapsed cube.
+        new_name:
+            New name for output cube; if None use iris default.
+
+    Returns:
+        Aggregated cube. If broadcast is True, dimensions are the same as input cube,
+        otherwise dimensions are collapsed."""
+
+    aggregator_dict = {
+        "sum": iris.analysis.SUM,
+        "mean": iris.analysis.MEAN,
+        "median": iris.analysis.MEDIAN,
+        "std_dev": iris.analysis.STD_DEV,
+        "min": iris.analysis.MIN,
+        "max": iris.analysis.MAX,
+    }
+
+    aggregator = aggregator_dict.get(aggregation)
+    if aggregator is None:
+        raise ValueError(
+            'aggregation must be one of "sum", "mean", "median", "std_dev", "min", "max"'
+        )
+
+    if isinstance(dimensions, str):
+        dimensions = [dimensions]
+    collapsed_cube = collapsed(cube, dimensions, aggregator)
+    if (
+        (aggregation == "std_dev")
+        and (collapsed_cube.data.size == cube.data.size)
+        and (np.ma.is_masked(collapsed_cube.data))
+    ):
+        # Standard deviation is undefined. Iris masks the entire output,
+        # but we also set the underlying data to np.nan here.
+        collapsed_cube.data.data[:] = np.nan
```

Comment:
The iris method returns masked data with value equal to `np.nan`. When this gets reshaped and combined into a single array in rainforests (in preparation for passing into the treelite predictor), the mask gets lost. So we need to ensure that the underlying data is also `np.nan`. Since iris returns masked data, I don't think there is any guarantee about the underlying value (although in practice it seems to be equal to `np.nan`). So I thought it was safest to set this explicitly.

Reply:
No worries. My initial comment wasn't entirely clear; it's not immediately clear that you are considering the case that the aggregation dimension(s) are of length 1. My suggestion was to change logical test from comparing size to checking the aggregation dimensions, or alternatively update the comment to explain when the standard deviation is undefined.

Now that the dimension being collapsed is "realization" only, it might be clearer to change this test to:
```
    if (
        (aggregation == "std_dev")
        and (cube.coord("realization").points.size == 1)
        and (np.ma.is_masked(collapsed_cube.data))
    ):
```

Reply:
done